go/libraries/doltcore/sqle/database.go

```diff
@@ -117,56 +125,114 @@ func (db *Database) GetTableInsensitive(ctx context.Context, tblName string) (sq
 		return NewLogTable(db.ddb, db.rsr), true, nil
 	}
 
-	tableNames, err := db.GetAllTableNames(ctx)
+	return db.getTable(ctx, db.root, tblName)
+}
 
+// GetTableInsensitiveAsOf implements sql.VersionedDatabase
+func (db *Database) GetTableInsensitiveAsOf(ctx *sql.Context, tableName string, time interface{}) (sql.Table, bool, error) {
+	root, err := db.rootAsOf(ctx, time)
 	if err != nil {
 		return nil, false, err
 	}
 
-	exactName, ok := sql.GetTableNameInsensitive(tblName, tableNames)
+	return db.getTable(ctx, root, tableName)
+}
 
+func (db *Database) rootAsOf(ctx *sql.Context, time interface{}) (*doltdb.RootValue, error) {
+	commit, ok := time.(string)
 	if !ok {
-		return nil, false, nil
+		panic("expected commit string")
+	}
+
+	cs, err := doltdb.NewCommitSpec(commit, "")
```

Comment:
if you want things like HEAD and HEAD^ to work then I believe you need the current branch.

Reply:
Teach me senpai